internal/pkg/agent/application/upgrade/step_unpack_test.go

```diff
@@ -481,8 +481,7 @@ func checkExtractedFilesOutOfOrder(t *testing.T, versionedHome string) {
 	}
```

Comment:
I thought you were going to handle unpacking in a follow up PR?  If so, please move the changes in this file to that PR. Again, as per our [contributing guidelines](https://github.com/elastic/elastic-agent/blob/main/CONTRIBUTING.md#as-a-reviewee-ie-author-of-a-pr), we should strive to keep our PRs atomic so they're easy to review.